source/isaaclab/isaaclab/envs/mdp/commands/force_pose_command.py

```diff
@@ -0,0 +1,169 @@
+from __future__ import annotations
+
+import torch
+from collections.abc import Sequence
+from typing import TYPE_CHECKING
+
+from isaaclab.assets import Articulation, RigidObject
+from isaaclab.sensors import ContactSensor
+from isaaclab.managers import CommandTerm
+from isaaclab.markers import VisualizationMarkers
+from isaaclab.utils.math import combine_frame_transforms, compute_pose_error, quat_from_euler_xyz, quat_unique, transform_points
+
+if TYPE_CHECKING:
+    from isaaclab.envs import ManagerBasedEnv
+
+    from .commands_cfg import UniformForcePoseCommandCfg
+
+
+class UniformForcePoseCommand(CommandTerm):
+    """Command generator for generating force-pose commands uniformly.
+
+    The command generator generates forces and poses by sampling positions uniformly within specified
+    regions in cartesian space in x, y while z is fixed to be on a surface. For orientation, it samples uniformly the euler angles
+    (roll-pitch-yaw) and converts them into quaternion representation (w, x, y, z). For forces, it samples uniformly
+    force setpoints within a predefined range.
+
+    The force, position and orientation commands are generated in the base frame of the robot, and not the
+    simulation world frame. This means that users need to handle the transformation from the
+    base frame to the simulation world frame themselves.
+
+    .. caution::
+
+        Sampling orientations uniformly is not strictly the same as sampling euler angles uniformly.
+        This is because rotations are defined by 3D non-Euclidean space, and the mapping
+        from euler angles to rotations is not one-to-one.
+
+    """
+
+    cfg: UniformForcePoseCommandCfg
+    """Configuration for the command generator."""
+
+    def __init__(self, cfg: UniformForcePoseCommandCfg, env: ManagerBasedEnv):
+        """Initialize the command generator class.
+
+        Args:
+            cfg: The configuration parameters for the command generator.
+            env: The environment object.
+        """
+        # initialize the base class
+        super().__init__(cfg, env)
+
+        # extract the robot and body index for which the command is generated
+        self.robot: Articulation = env.scene[cfg.asset_name]
+        self.surface: RigidObject = env.scene[cfg.surface_name]
+        self.force_sensor: ContactSensor = env.scene[cfg.force_sensor_name]
+        self.body_idx = self.robot.find_bodies(cfg.body_name)[0][0]
+
+        # create buffers
+        # -- commands: (x, y, z, qw, qx, qy, qz, fx, fy, fz) in root frame
+        self.pose_command_b = torch.zeros(self.num_envs, 10, device=self.device)
+        self.pose_command_b[:, 3] = 1.0
+        self.pose_command_w = torch.zeros_like(self.pose_command_b)
```

Comment:
I would prefer either `command` or `pose_force_command` - just `pose_command` is misleading. The class name is also "ForcePose" even though force comes at the end of the vector, maybe that could be changed to "PoseForce"